### PR TITLE
Fix setup.py dependencies, link, and format

### PR DESCRIPTION
**Pull request by VladimirSlavik**
Could this fix the dependencies problem #27?

### CODE REVIEW

**Comment by VladimirSlavik** [comment]
It might be a good idea to change the maintainer or at least email too...

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 setup(name='python-meh', version='0.48',
       description='Python module for handling exceptions',
       author='Chris Lumens', author_email='clumens@redhat.com',
-      url='http://git.fedoraproject.org/git/?p=python-meh.git',
-      data_files = [('/usr/share/python-meh', ['ui/exception-dialog.glade'])],
-      packages=['meh', 'meh.ui'])
+      url='https://github.com/rhinstaller/python-meh',
+      data_files=[('/usr/share/python-meh', ['ui/exception-dialog.glade'])],
+      packages=['meh', 'meh.ui'],
+      install_requires=['report', 'pydbus'])
```
Comment (jkonecny12) [inline setup.py]: I think there should me more requirements than just these two based on spec file.

Comment (VladimirSlavik) [reply setup.py]: Changed - now there's everything (I believe).

**Review by jkonecny12 — APPROVED** [review]
I didn't tested it but seems good to me.

**Comment by VladimirSlavik** [comment]
Hi @kloczek , can you please check if this fixes what you asked?

**Comment by kloczek** [comment]
OK one sec :)

**Comment by kloczek** [comment]
```
      install_requires=['dbus-python', 'PyGObject', 'report', 'rpm'])
```
That line dowes not adds any dependencies to egg-info files and it is install time dependency and these are not python pip dependencies but rpm packages names.
https://packaging.python.org/discussions/install-requires-vs-requirements/

```
$ for i in dbus-python PyGObject report rpm; do rpm -q --whatprovides python3dist\($i\); done
python3-dbus-1.2.16-1.fc33.x86_64
no package provides python3dist(PyGObject)
no package provides python3dist(report)
python-rpm-4.15.1-3.fc33.x86_64
````
Correct dependencies should add to egg-info dependencies for  dbus-python and report dependencies.

Here is list of provided dependencies by dbus python module:
````
]$ rpm -q --qf "[%{PROVIDES} %{PROVIDEFLAGS:depflags} %{PROVIDEVERSION}\n]\n" python3-dbus
python-dbus = 1.2.16-1.fc33
python3-dbus = 1.2.16-1.fc33
python3-dbus(x86-64) = 1.2.16-1.fc33
python3.8dist(dbus-python) = 1.2.16
python3dist(dbus-python) = 1.2.16               <<=== here that can be used
````
report python package does not have even egg-info  description.

**Comment by kloczek** [comment]
Hrere is list of modules which needs meh:
```
[tkloczko@barrel python-meh-0.48]$ grep -r ^import -w | awk -F: '{print $2}'|sort| uniq
import bdb
import copy
import dbus
import gettext
import gi
import glob
import importlib
import inspect
import meh
import meh.dump
import meh.handler
import os
import report
import report.io.TextIO
import subprocess
import sys
import tempfile
import traceback
import unittest
```
Some does not need to be listed in dependencies as they are part of the python core.

**Comment by kloczek** [comment]
Below output woud be better on assesing which one modules shoud be added to esternal runtime dependencies:
```
[tkloczko@barrel python-meh-0.48]$ grep -r import -w | awk -F: '{print $2}'|sort| uniq| grep -v ^from
        from inspect import stack as _stack
import bdb
import copy
import dbus
import gettext
import gi
import glob
        import hashlib
import importlib
import inspect
import meh
import meh.dump
import meh.handler
        import meh.ui.gui
        import meh.ui.text
import os
        import pdb
import report
import report.io.TextIO
            import rpm
import subprocess
import sys
        import tempfile
import tempfile
import traceback
import unittest
    Move "import rpm" to where it's needed to avoid nameserver problems.
- Move "import rpm" to where it's needed to avoid nameserver problems (clumens).
    Pull in a missing import.
    remove unused import subprocess
    Use gi.require_version for Gtk import
- Use gi.require_version for Gtk import (vtrefny)
```

**Comment by VladimirSlavik** [comment]
So... based on the above, clearly I don't know what I am doing :-/ I'll have to learn more about setup and distutils and that stuff.

**Comment by kloczek** [comment]
This is what I have now as static (not autogenerated python dependency:
```
Requires:       python3dist(dbus-python)
```
So definitelly dbus module on runtime taht dependency must be added.

**Comment by VladimirSlavik** [comment]
A few months later, it's clear I can't find time to actually learn this. Thus closing. The prettifying parts are now in #30, so only the dependencies issue remains.

**Pull request closed without merging**